src/components/common/PeripheralsDialog.vue

```diff
@@ -0,0 +1,428 @@
+<template>
+  <app-dialog
+    v-model="open"
+    :title="$t('app.system_info.label.devices')"
+    max-width="550"
+    no-actions
+  >
+    <v-toolbar dense>
+      <v-tabs
+        v-model="tab"
+        show-arrows
+      >
+        <v-tab
+          v-for="peripheralGroup in peripheralGroups"
+          :key="peripheralGroup.type"
+        >
+          <div>
+            {{ $filters.startCase(peripheralGroup.type) }}
+            <v-chip
+              small
+              link
+              class="ml-2"
+            >
+              {{ peripheralGroup.count ?? '?' }}
+            </v-chip>
+          </div>
+        </v-tab>
+      </v-tabs>
+
+      <v-spacer />
+
+      <v-tooltip bottom>
+        <template #activator="{ on, attrs }">
+          <v-btn
+            v-bind="attrs"
+            fab
+            text
+            small
+            :disabled="!currentPeripheralGroup || hasWaitFor(currentPeripheralGroup)"
+            @click.prevent.stop="handleRefresh()"
+            v-on="on"
+          >
+            <v-icon
+              :class="{ 'spin-alt': currentPeripheralGroup && hasWaitFor(currentPeripheralGroup) }"
+            >
+              $refresh
+            </v-icon>
+          </v-btn>
+        </template>
+        <span>{{ $t('app.general.btn.refresh') }}</span>
+      </v-tooltip>
+    </v-toolbar>
+
+    <v-card-text>
+      <v-tabs-items
+        v-model="tab"
+        touchless
+      >
+        <v-tab-item
+          v-for="peripheralGroup in peripheralGroups"
+          :key="peripheralGroup.type"
+        >
+          <template v-if="peripheralGroup.count == null">
+            Click on the refresh button to search for devices
```

Comment:
I think this should be extracted into the language definitions

Reply:
I'm also wondering if we should just trigger the device search automatically when a tab is selected and there's no data for it yet, that would save the user a couple clicks

Reply:
> I think this should be extracted into the language definitions

Nice catch, that is now fixed!

> I'm also wondering if we should just trigger the device search automatically when a tab is selected and there's no data for it yet, that would save the user a couple clicks

I gave this a try, and though it works it is so fast that causes a weird flash and then a resize of the dialog. There is also the danger for CAN as from what I read, this should better be left as a manual process (single node ok, multi-node is bad!).

For now, I will leave this to the user to click "refresh".